std/variant.d

```diff
@@ -1608,6 +1615,41 @@ pure nothrow @nogc
     assert(v != b);
 }
 
+// Can compare with 'null'
+@system unittest
+{
+    static struct Bar
+    {
+        int* ptr;
+        alias ptr this;
+    }
+
+    static class Foo {}
+    int* iptr;
+    int[] arr;
+
+    Variant v = Foo.init; // 'null'
+    assert(v != null); // can only compare objects with 'null' by using 'is'
+
+    v = iptr;
+    assert(v == null); // pointers can be compared with 'null'
+    assert(v == iptr);
+
+    v = arr;
+    assert(v == null); // arrays can be compared with 'null'
+    assert(v == arr);
```

Comment:
According to codecov, it is irrelevant to add such non-null comparison asserts. There is already tests covering this behaviour, e.g. in 1328. See https://app.codecov.io/gh/dlang/phobos/blob/master/std/variant.d

Reply:
At first, I was having issues with my unittests not passing *only* with *my* non-null test, it was for debugging only and I had the logic wrong at the time, but I decided to leave it because of that. I can remove it if it matters.

Reply:
done

Reply:
I read this wrong, I read as `non-equal to null` tests instead of what you actually meant.